Sources/Classes/Views/BigBossButton/BigBossButtonView.swift

```diff
@@ -0,0 +1,379 @@
+    //                var state = self.stateRelay.value
+    //                state.subtract([.enabled, .disabled])
+    //                state.insert(isEnabled ? .enabled : .disabled)
+    //                self.stateRelay.accept(state)
+    //            }
+    //        }
+    //    }
+    //
+    //    private func observeIsHighlighted() -> NSKeyValueObservation {
+    //        return button.observe(\BigBossButton.isHighlighted, options: .new,
+    //      changeHandler: { [weak self] _, isHighlighted in
+    //
+    //            guard let self = self else {
+    //                return
+    //            }
+    //
+    //            if let isHighlighted = isHighlighted.newValue {
+    //                var state = self.stateRelay.value
+    //                state.subtract([.highlighted, .unhighlighted])
+    //                state.insert(isHighlighted ? .highlighted : .unhighlighted)
+    //                self.stateRelay.accept(state)
+    //            }
+    //        })
+    //    }
+
+    // MARK: - UI
+
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        shadowView.layer.shadowPath = UIBezierPath(rect: button.bounds).cgPath
+    }
+
+    public var buttonIsDisabledWhileLoading = false
+
+    private func set(active: Bool) {
+        button.isEnabled = buttonIsDisabledWhileLoading ? !active : true
+        if active {
+            spinnerView?.isHidden = false
+            spinnerView?.startAnimating()
+        } else {
+            spinnerView?.isHidden = true
+            spinnerView?.stopAnimating()
+        }
+    }
+
+    private func addSpinner() {
+        if let spinner = spinnerView {
+            addSubview(spinner)
+            configureSpinnerConstraints()
+        }
+    }
+
+    private func removeSpinner() {
+        if spinnerView != nil {
+            self.spinnerView?.removeFromSuperview()
+            self.spinnerView = nil
+        }
+    }
+
+    // MARK: - Layout
+
+    override open var forFirstBaselineLayout: UIView {
+        return button.forFirstBaselineLayout
+    }
+
+    override open var forLastBaselineLayout: UIView {
+        return button.forLastBaselineLayout
+    }
+
+    private func configureConstraints() {
+        button.constaintToEdges(of: self, with: appearance.buttonInsets)
+        configureShadowViewConstraints()
+    }
+
+    private func configureSpinnerConstraints() {
+        switch appearance.spinnerPosition {
+        case .center:
+            spinnerView?.centerXAnchor.constraint(equalTo: button.centerXAnchor).isActive = true
+            spinnerView?.centerYAnchor.constraint(equalTo: button.centerYAnchor).isActive = true
+        case .leftToText(let offset):
+            if let buttonLabel = button.titleLabel {
+                spinnerView?.centerYAnchor.constraint(equalTo: buttonLabel.centerYAnchor).isActive = true
+                spinnerView?.trailingAnchor.constraint(equalTo: buttonLabel.leadingAnchor,
```

Comment:
is spinner and text centered together in button?

Reply:
yes